ktlint-ruleset-standard/src/test/kotlin/com/pinterest/ktlint/ruleset/standard/TrailingCommaRuleTest.kt

```diff
@@ -1021,4 +1021,157 @@ class TrailingCommaRuleTest {
                 LintViolation(7, 6, "Missing trailing comma before \")\"")
             ).isFormattedAs(formattedCode)
     }
+
+    @Test
+    fun `Given that a trailing comma is required then add trailing comma after last enum member`() {
+        val code =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE,
+            }
+            """.trimIndent()
+
+        trailingCommaRuleAssertThat(code)
+            .withEditorConfigOverride(allowTrailingCommaProperty to true)
+            .hasLintViolation(3, 13, "Missing trailing comma before \"}\"")
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Given that a trailing comma is is not allowed then remove comma after last enum member`() {
+        val code =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE,
+            }
+            """.trimIndent()
+        val formattedCode =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE
+            }
+            """.trimIndent()
+
+        trailingCommaRuleAssertThat(code)
+            .withEditorConfigOverride(allowTrailingCommaProperty to false)
+            .hasLintViolation(3, 13, "Unnecessary trailing comma before \"}\"")
+            .isFormattedAs(formattedCode)
+    }
+
+    @Test
+    fun `Given that a trailing comma is not allowed then nothing happens for enums terminated with semicolon`() {
+        val code =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE;
+
+                fun print() = name()
+            }
+            """.trimIndent()
+
+        trailingCommaRuleAssertThat(code)
+            .withEditorConfigOverride(allowTrailingCommaProperty to false)
+            .hasNoLintViolations()
+    }
+
+    @Test
+    fun `Given that a trailing comma is allowed then nothing happens for enums terminated with semicolon`() {
+        val code =
+            """
+            enum class Shape {
+                SQUARE,
+                TRIANGLE;
+
+                fun print() = name()
+            }
+            """.trimIndent()
+
+        trailingCommaRuleAssertThat(code)
+            .withEditorConfigOverride(allowTrailingCommaProperty to true)
+            .hasNoLintViolations()
+    }
+
+    @Test
+    fun `Given that a trailing comma is allowed then it is added for complicated enums`() {
```

Comment:
Check all your tests for wording. It should either be "is not allowed" or "is required". "is allowed" is misleading as it does sound as something that is optionally allowed instead of being required.

Reply:
I agree. I thought I saw the "given - is allowed" wording elsewhere in tests but I must've had it mixed up. Will fix.